File: 2020/Day3Test.cs
using System;
using NUnit.Framework;

namespace AoC.Tests
{
    [TestFixture]
    class Day3Test
    {
        public static readonly string[] INPUT = { 
            "..##.......",
            "#...#...#..",
            ".#....#..#.",
            "..#.#...#.#",
            ".#...##..#.",
            "..#.##.....",
            ".#.#.#....#",
            ".#........#",
            "#.##...#...",
            "#...##....#",
            ".#..#...#.#",
            "..........."    // added an extra line to weed out x/y confusion related bugs that a square map hides
       };
 
        private IPuzzle puzzle;

        public Day3Test()
        {
            this.puzzle = new Day3();
        }

        [TestCase]
        public void TestPart1()
        {   
            Assert.AreEqual("7", this.puzzle.Part1(INPUT));
        }
        
        [TestCase]
        public void TestPart2()
        {   
            Assert.AreEqual("336", this.puzzle.Part2(INPUT));
        }
    }
}

File: aoc2025/cmd/day3/day3_test.go
package main

import (
	"testing"
)

var input = `987654321111111
811111111111119
234234234234278
818181911112111`

func TestPart1(t *testing.T) {
	actual := Part1(input)
	expected := "357"
	if actual != expected {
		t.Errorf(`Part1(input) = %q, want %q`, actual, expected)
	}
}
func TestPart2(t *testing.T) {
	actual := Part2(input)
	expected := "3121910778619"
	if actual != expected {
		t.Errorf(`Part2(input) = %q, want %q`, actual, expected)
	}
}
